selection_sort.go

```diff
@@ -0,0 +1,16 @@
+package sort
+
+func selectionSort(arr []int) []int {
+	for i := 0; i < len(arr)-1; i++ {
+		minIdx := i
+		for j := i; j < len(arr); j++ {
```

Comment:
### Optimization
The inner loop can be optimized by starting from `i + 1` instead of `i`, as the elements before `i` are already sorted.

**Suggested Fix:**
```go
for j := i + 1; j < len(arr); j++ {
```